services.py

```diff
@@ -1,44 +1,32 @@
-import os
+import json
 import requests
 import subprocess
 import configparser
 import geocoder
 import ast
 import logging
-from geopy.geocoders import Nominatim
+from geopy.geocoders import Photon
+import serial
+import darkdetect
+import screen_brightness_control as sbc
+
+
+SERIAL_PORT = "/dev/ttyUSB0"
+
+
 
 class Services:
     def __init__(self):
-        self.logger = self.get_logger('services')
+        self.logger = logging.getLogger('services')
         self.formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
 
-    def run_bash(self, script_path):
-        try:
-            subprocess.run(["bash", script_path], capture_output=True)
-            self.logger.info('Script executed successfully')
-        except subprocess.CalledProcessError as e:
-            self.logger.error(f'Error running bash script: {e}')
-
-    def run_python(self, script_path):
-        try:
-            with open(script_path, 'r') as f:
-                script = f.read()
-            exec(script)
-            self.logger.info('Script executed successfully')
-        except Exception as e:
-            self.logger.error(f'Error running python script: {e}')
-            
     def fetch_json(self, url):
         response = requests.get(url)
-        if (response.status_code != 204 and response.status_code < 300 and
+        if not (response.status_code != 204 and response.status_code < 300 and
                 response.headers["content-type"].strip().startswith("application/json")):
-            try:
-                return response.json()
-            except ValueError:
-                self.logger.critical(f'Error loading JSON: {response.status_code}')
-                return response.status_code
-        self.logger.critical(f'Error loading JSON: {response.status_code}')
-        return None
+            self.logger.error(f'Error loading json: {response.status_code}')
+            raise RuntimeError(f'Error loading json: {response.status_code}')
```

Comment:
if you are logging the `RuntimeError` at the place where you catch it, no need to also log it here.
In general, you usually do only at most one of the following in every given place in the code:
1. log which operation starts
2. log which operation succeeded
3. log which operation failed WHEN the failure was not passed over up the stack
4. raise or pass over the error without logging it

Options 1 may technically be present in the same place with options 2 or 3 when you want to report the start of the next action right after the (successful or unsuccessful) finish of the previous action, but as you can see, conceptually these a still 2 different places.